src/vs/workbench/contrib/scm/browser/scmViewService.ts

```diff
@@ -108,32 +116,51 @@ export class SCMViewService implements ISCMViewService {
 			}, 0)
 	);
 
-	private _focusedRepository: ISCMRepository | undefined;
-
 	get focusedRepository(): ISCMRepository | undefined {
-		return this._focusedRepository;
+		return this._repositories.find(r => r.focused)?.repository;
 	}
 
 	private _onDidFocusRepository = new Emitter<ISCMRepository | undefined>();
 	readonly onDidFocusRepository = this._onDidFocusRepository.event;
 
-	private _compareRepositories: (op1: ISCMRepository, op2: ISCMRepository) => number;
+	private _repositoriesSortKey: ISCMRepositoryViewSortKey;
+	private _compareRepositories: (op1: ISCMRepositoryView, op2: ISCMRepositoryView) => number;
 
 	constructor(
-		@ISCMService private readonly scmService: ISCMService,
+		@ISCMService scmService: ISCMService,
 		@IInstantiationService instantiationService: IInstantiationService,
 		@IStorageService private readonly storageService: IStorageService,
-		@IWorkspaceContextService workspaceContextService: IWorkspaceContextService
+		@IWorkspaceContextService workspaceContextService: IWorkspaceContextService,
+		@IConfigurationService configurationService: IConfigurationService,
 	) {
 		this.menus = instantiationService.createInstance(SCMMenus);
 
-		this._compareRepositories = (op1: ISCMRepository, op2: ISCMRepository): number => {
-			const name1 = getRepositoryName(workspaceContextService, op1);
-			const name2 = getRepositoryName(workspaceContextService, op2);
+		try {
+			this.previousState = JSON.parse(storageService.get('scm:view:visibleRepositories', StorageScope.WORKSPACE, ''));
+		} catch {
+			// noop
+		}
+
+		this._repositoriesSortKey = this.previousState?.sortKey ?? configurationService.getValue('scm.repositories.sortOrder');
+
+		this._compareRepositories = (op1: ISCMRepositoryView, op2: ISCMRepositoryView): number => {
```

Comment:
Let's just make this a proper `private` method of the class, makes things easier to reason about. Make sure to use `bind(this)` as in `this._repositories.sort(this._compareRepositories.bind(this));` when moving it to its own method.

Reply:
Fixed.